src/common/procedure/src/local.rs

```diff
@@ -368,38 +369,70 @@ pub struct LocalManager {
     procedure_store: Arc<ProcedureStore>,
     max_retry_times: usize,
     retry_delay: Duration,
-    remove_outdated_meta_task: RepeatedTask<Error>,
+    state: Arc<Mutex<LocalManagerState>>,
+    running: Arc<AtomicBool>,
+    config: ManagerConfig,
+}
+
+#[derive(Debug, Default)]
+pub struct LocalManagerState {
```

Comment:
Maybe `BackgroundTasks` is more proper. `ManagerContext` is closer to the state of the manager.